specs/create-rbd-spec.adoc

```diff
@@ -0,0 +1,236 @@
+= Create RBD workflow by using tendrl-frontend UI
+
+The intent of create RBD workflow is to provide a mechanism to create one or more Ceph RBDs into tendrl for management using tendrl-frontend UI.
+
+* Assumption: 
+
+    - Ceph cluster exists.
+
+    - Ceph pool in the Ceph cluster may or may exist.
+
+== Problem description
+
+This specification describes a mechanism to create one or more Ceph RBDs into tendrl for management.
+
+== Use Cases
+
+This addresses the use case of creating RBD into tendrl for management.
+
+The workflow is as follows:
+
+* "Create RBD" is triggered by "Create RBD" button on RBD landing page.
+   (https://redhat.invisionapp.com/share/BR8JDCGSQ#/screens/198417119)
+
+* User clicks on "Create" button.
+
+* It will redirect the user to a wizard for creating RBD/RBDs.
+
+* The flow consist of three steps.
+
+* In the first step, user needs to fill below items to the form -
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735641)
+
+    - RBD Name
+
+    - RBDs to create(UI will autogenerate the name of RBDs based on the number)
+
+    - cluster(User will select from the drop down which will list all the clusters present in Tendrl)
+
+    - Target size
+
+* If the target size * RBDs to create > available storage in this cluster, a warning message will be displayed, but 
+  the user will be allowed to continue.
+
+* For block devices, a backing pool must be created. Only one pool can be created per set of block devices.
+
+* In the second step, user will decide whether to use an existing pool or create a new pool as the backing pool
+  for this set of RBD.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735638)
+
+* On choosing existing pool, a drop down will be displayed which will list down all the available pools in the 
+  Tendrl.
+
+* If using an existing pool, the total storage required by all the RBDs must not exceed the available capacity of  
+  the chosen cluster(i.e. Target size * #RBDs to create <= available capacity in cluster).
+
+* When user selects an existing pool, the attributed of that pool will be shown along with a graphic that reflects 
+  storage currently in use by that pool, the amount to be added, and the amount remaining in the cluster after the creation of the new RBD storage.
+
+* The details to be shown on the UI on selecting existing pool are -
+
+    - Type
+
+    - OSDs
+
+    - Replicas
+
+    - PG Count
+
+    - Journal Configuration
+
+    - Quotas
+
+* If user wants to create a new pool, they are presented with a pool creation form. See Create Pool Workflow's spec 
+  for details about these options.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735639)
```

Comment:
@anivargi So first the UI needs to send the POST call for creating pool in the 2nd step and after that only POST call for creating RBD will be sent in the last step. Please confirm whether my assumption is correct or not.